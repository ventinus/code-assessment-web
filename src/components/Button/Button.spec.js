import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import {
  PrimaryButton,
  SecondaryButton,
  SecondaryAltButton,
  DecButton,
  IncButton
} from '.'

const buttons = [
  PrimaryButton,
  SecondaryButton,
  SecondaryAltButton,
  DecButton,
  IncButton
]

const setup = (Comp, props) => {
  const component = mount(<Comp {...props}>Button Text</Comp>)

  return {
    component,
    button: component.find('button'),
  }
}

let fn

describe('Button components', () => {
  beforeEach(() => {
    fn = jest.fn()
  })

  buttons.forEach(Btn => {
    it(`renders correctly ${Btn.name}`, () => {
      const tree = renderer
        .create(<Btn onClick={jest.fn()}>children</Btn>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`passes children as text ${Btn.name}`, () => {
      const { component } = setup(Btn, {onClick: fn})
      expect(component.text()).toBe('Button Text')
    })

    it(`sets up onClick callbacks ${Btn.name}`, () => {
      const { button } = setup(Btn, {onClick: fn})
      button.simulate('click')
      expect(fn).toHaveBeenCalled()
    })

    it(`prevents onClick when disabled ${Btn.name}`, () => {
      const { button } = setup(Btn, {onClick: fn, disabled: true})
      button.simulate('click')
      expect(fn).not.toHaveBeenCalled()
    })
  })
})
