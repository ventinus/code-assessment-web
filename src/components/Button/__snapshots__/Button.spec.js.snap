// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components renders correctly DecButton 1`] = `
<button
  className="button type--a8  button--secondary button--dec"
  disabled=""
  onClick={[Function]}
  type="button"
>
  <span
    className="is-visually-hidden"
  >
    children
  </span>
</button>
`;

exports[`Button components renders correctly IncButton 1`] = `
<button
  className="button type--a8  button--secondary-alt button--inc"
  disabled=""
  onClick={[Function]}
  type="button"
>
  <span
    className="is-visually-hidden"
  >
    children
  </span>
</button>
`;

exports[`Button components renders correctly PrimaryButton 1`] = `
<button
  className="button type--a8  button--primary "
  disabled=""
  onClick={[Function]}
  type="button"
>
  children
</button>
`;

exports[`Button components renders correctly SecondaryAltButton 1`] = `
<button
  className="button type--a8  button--secondary-alt "
  disabled=""
  onClick={[Function]}
  type="button"
>
  children
</button>
`;

exports[`Button components renders correctly SecondaryButton 1`] = `
<button
  className="button type--a8  button--secondary "
  disabled=""
  onClick={[Function]}
  type="button"
>
  children
</button>
`;
